src/libraries/Microsoft.PowerFx.Interpreter/Functions/LibraryText.cs

```diff
@@ -1196,5 +1199,224 @@ internal static bool TryGetInt(FormulaValue value, out int outputValue)
+                {
+                    _flags.IgnoreBinaryData = arg1string.Value.Contains("G");
+                    _flags.IgnoreUnsupportedTypes = arg1string.Value.Contains("I");
+                    _flags.IncludeBinaryData = arg1string.Value.Contains("B");
+                    _flags.IndentFour = arg1string.Value.Contains("4");
+                }
+
+                if (_args.Length > 1 && _args[1] is OptionSetValue arg1optionset)
+                {
+                    _flags.IgnoreBinaryData = arg1optionset.Option == "IgnoreBinaryData";
+                    _flags.IgnoreUnsupportedTypes = arg1optionset.Option == "IgnoreUnsupportedTypes";
+                    _flags.IncludeBinaryData = arg1optionset.Option == "IncludeBinaryData";
+                    _flags.IndentFour = arg1optionset.Option == "IndentFour";
+                }
+
+                _flags.Compact = !_flags.IndentFour;
+
+                if ((_flags.IncludeBinaryData && _flags.IgnoreBinaryData) ||
+                    (_flags.HasMedia && !_flags.IncludeBinaryData && !_flags.IgnoreBinaryData) ||
+                    (!_flags.IgnoreUnsupportedTypes && JsonFunction.HasUnsupportedType(_args[0].Type._type, out var _, out var _)))
+                {
+                    _flags = null;
+                    return;
+                }
+            }
+
+#pragma warning disable CS0618 // Use of [Obsolete] member: Date[Time]Value.Value
+
+            internal string JsonInternal(FormulaValue arg0, int indent)
+            {
+                return arg0 switch
+                {
+                    BlankValue blkv => "null",
+                    BooleanValue bv => bv.Value.ToString(CultureInfo.InvariantCulture).ToLowerInvariant(),
+                    ColorValue cv => GetColorString(cv.Value),
+                    DateTimeValue dtv => WithQuotes(TimeZoneInfo.ConvertTimeToUtc(dtv.Value, _timeZoneInfo).ToString("yyyy-MM-dd'T'HH:mm:ss.fffK", CultureInfo.InvariantCulture)),
+                    DateValue dv => WithQuotes(TimeZoneInfo.ConvertTimeToUtc(dv.Value, _timeZoneInfo).ToString("yyyy-MM-dd", CultureInfo.InvariantCulture)),
+                    DecimalValue dmv => dmv.Value.ToString(CultureInfo.InvariantCulture),
+                    GuidValue gv => WithQuotes(gv.Value.ToString("D", CultureInfo.InvariantCulture)),
+                    NumberValue nv => nv.Value.ToString(CultureInfo.InvariantCulture),
+                    OptionSetValue osv => GetOptionSetValueString(osv),
+                    RecordValue rv => GetJsonStringForRecord(rv, indent),
+                    StringValue sv => WithQuotes(sv.Value),
+                    TableValue tv => GetJsonStringForTable(tv, indent),
+                    TimeValue tmv => WithQuotes(tmv.Value.ToString(@"hh\:mm\:ss\.fff", CultureInfo.InvariantCulture)),
+                    _ => "invalid",
+                };
+            }
+
+#pragma warning restore CS0618
+
+            private static string WithQuotes(string str) => $@"""{str}""";

```

Comment:
>$@"""{str}""" [](http://example.com/codeflow?start=60&length=13)

does this need to escape?

Reply:
Carlos mentionned it, this needs to be fixed